typegraph/core/src/utils/fs_host.rs

```diff
@@ -214,3 +216,64 @@ pub fn hash_file(path: &Path) -> Result<(String, u32), String> {
 pub fn path_exists(path: &Path) -> Result<bool, String> {
     path_exists_host(&path.to_string_lossy())
 }
+
+pub fn is_glob(path: &str) -> bool {
+    let path = PathBuf::from(path);
+    if let Some(base_name) = path.file_name() {
+        let base_name = base_name.to_str().unwrap();
+        if base_name.contains('*') || base_name.contains('?') {
+            return true;
+        }
+    }
+
+    false
+}
+
+pub fn get_dir(path: &str) -> PathBuf {
+    let directory = PathBuf::from(path);
+
+    PathBuf::from(directory.parent().unwrap())
+}
+
+pub fn get_matching_files(glob_pattern: &str) -> Result<Vec<PathBuf>, String> {
+    let mut matching_files = vec![];
+    let glob_dir = get_dir(glob_pattern);
+
+    let glob_abs_path = make_absolute(&glob_dir)?;
+    let all_files = expand_path(&glob_abs_path, &[])?;
+    let glob_pattern = GlobPattern::new(glob_pattern).unwrap();
+    // eprint(&format!("++++++++++++++++++ {}", glob_pattern.as_str()));
+    for file in all_files {
+        if glob_pattern.matches(file.to_str().unwrap()) {
+            matching_files.push(make_relative(&file)?);
+        }
+    }
+
+    // eprint(&format!("***************** {:?}", matching_files));
+    Ok(matching_files)
+}
+
+pub fn resolve_globs_dirs(deps: Vec<String>) -> Result<Vec<PathBuf>, String> {
+    let mut resolved_deps = vec![];
+    for dep in deps {
+        if is_glob(&dep) {
+            let _ = get_matching_files(&dep)
+                .unwrap()
+                .into_iter()
+                .map(|file| resolved_deps.push(file));
+        } else {
+            // print(&format!(
+            //     "***************** {:?}",
+            //     make_absolute(&PathBuf::from(dep.clone()))?
```

Comment:
remove the debug comment or can we have log with given level in wasm?

Reply:
I will remove the all debug comments in the end all at once. I haven't found any loggers in the core SDK but I will check if that's possible.